### ISSUE TITLE
Run faster! Compile and move the proto plugin

### ISSUE THREAD

**Issue by jtmcdole** [pikaju/dart-protoc-builder#7]
reference: https://github.com/pikaju/dart-protoc-builder/blob/90966ef0f1fd38ec56dcec9b42710d11d3943c43/lib/src/protoc_plugin_download.dart#L49-L51

The plugin's "executable" is just running dart (the vm) - loading and parsing each and every execution. I've noticed this to be a significant slowdown.

```dart
    // Compile the program for faster startup.
    await Process.run('dart', ['compile', 'exe', 'bin/protoc_plugin.dart'],
        workingDirectory: protocPluginPackageDirectory.path);

    // Move the program so no one can tell any better....
    // dart compile always writes ".exe" file on every system.
    final protocBin = '${protocPluginPackageDirectory.path/bin/protoc_plugin.exe';
    await File(protocBin).rename(protocPlugin .path);
```

**Comment by jtmcdole** [comment]
sort-of-proof (I'm working on a "also compile all dependencies" concept):

dart-snapshot version (already installed in pubcache). This one is many many times faster than the default "dart run" version in this build:
[![asciicast](https://asciinema.org/a/511361.svg)](https://asciinema.org/a/511361)

However, compiling to exe makes this tremendously faster:
[![asciicast](https://asciinema.org/a/511381.svg)](https://asciinema.org/a/511381)

**Comment by jtmcdole** [comment]
I have implemented this in a non-builder command line tool: https://github.com/jtmcdole/protofu  Feel free to copy whatever you like.

**Comment by pikaju** [comment]
Thank you for the suggestion, I implemented this for version `0.4.0` (better late than never eh?)

**Comment by jtmcdole** [comment]
🎉 Any improvement is always a huge win.
